### PR TITLE
Problem: cli keyring-backend not working correctly (fix #724)

### LINKED ISSUE #724: Problem: config/client.toml keyring-backend  not working correctly
if modify like this
```
keyring-backend = "file"
```

default keyring backend is still os, in adding key

### PR DESCRIPTION
**Pull request by leejw51crypto**
in adding key,
config/client.toml backend is not used 

1. adding key <- config/client.toml not used
2. list key <- config/client.toml used

### CODE REVIEW

File: client/keys.go [hunk 1]
```diff
@@ -91,16 +91,15 @@ func runAddCmd(cmd *cobra.Command, args []string) error {
 	)
 
 	dryRun, _ := cmd.Flags().GetBool(flags.FlagDryRun)
+	newcontext := clientCtx
 	if dryRun {
 		kr, err = keyring.New(sdk.KeyringServiceName(), keyring.BackendMemory, clientCtx.KeyringDir, buf, hd.EthSecp256k1Option())
-	} else {
-		backend, _ := cmd.Flags().GetString(flags.FlagKeyringBackend)
-		kr, err = keyring.New(sdk.KeyringServiceName(), backend, clientCtx.KeyringDir, buf, hd.EthSecp256k1Option())
+		newcontext = clientCtx.WithKeyring(kr)
```
Comment (khoslaventures) [inline client/keys.go:97]: you can just update with `clientCtx = clientCtx.WithKeyring(kr)`, no need for `newcontext`

File: client/keys.go [hunk 2]
```diff
@@ -91,16 +91,15 @@ func runAddCmd(cmd *cobra.Command, args []string) error {
...
 	}
 
 	if err != nil {
 		return err
 	}
 
-	return clientkeys.RunAddCmd(clientCtx.WithKeyring(kr), cmd, args, buf)
+	return clientkeys.RunAddCmd(newcontext, cmd, args, buf)
```
Comment (khoslaventures) [inline client/keys.go:104]: use an updated clientCtx

File: client/keys.go (hunk 1 above)
Comment (leejw51crypto) [reply client/keys.go:97]: that's overrides initial keyring setup in client.toml
so i think for dryrun, use current paradigm ?

File: client/keys.go (hunk 2 above)
Comment (leejw51crypto) [reply client/keys.go:104]: it's for just using original keyring backend which is from client.toml
(possible modified by flag --keyring-backend)

without this code, user initial setting is not used

File: client/keys.go (hunk 1 above)
Comment (fedekunze) [inline client/keys.go:97]: This only overrides the keyring 👍

```suggestion
		clientCtx = clientCtx.WithKeyring(kr)
```

File: client/keys.go (hunk 2 above)
Comment (fedekunze) [inline client/keys.go:104]:
```suggestion
	return clientkeys.RunAddCmd(clientCtx, cmd, args, buf)
```

**Review by fedekunze — COMMENTED** [review]
Thanks! LGTM, can you add a bug fix changelog entry too?

File: client/keys.go [hunk 3]
```diff
@@ -91,16 +91,15 @@ func runAddCmd(cmd *cobra.Command, args []string) error {
 	)
 
 	dryRun, _ := cmd.Flags().GetBool(flags.FlagDryRun)
+	newcontext := clientCtx
```
Comment (fedekunze) [inline client/keys.go:94]:
```suggestion
```

**Comment by fedekunze** [comment]
@leejw51crypto bump

**Comment by leejw51crypto** [comment]
i'll rebase and add to changelog

**Comment by leejw51crypto** [comment]
done

**Pull request merged**